inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/utils/MSSQLServerContainer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sort.tests.utils;
+
+import com.google.common.collect.Sets;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.LicenseAcceptance;
+
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class MSSQLServerContainer extends JdbcDatabaseContainer {
+
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("mcr.microsoft.com/mssql/server");
+    /** @deprecated */
+    @Deprecated
+    public static final String DEFAULT_TAG = "2017-CU12";
+    public static final String NAME = "sqlserver";
+    public static final String IMAGE;
+    public static final Integer MS_SQL_SERVER_PORT;
+    static final String DEFAULT_USER = "SA";
+    static final String DEFAULT_PASSWORD = "A_Str0ng_Required_Password";
+    private String password;
+    private static final int DEFAULT_STARTUP_TIMEOUT_SECONDS = 240;
+    private static final int DEFAULT_CONNECT_TIMEOUT_SECONDS = 240;
+    private static final Pattern[] PASSWORD_CATEGORY_VALIDATION_PATTERNS;
+
+    /** @deprecated */
+    @Deprecated
+    public MSSQLServerContainer() {
+        this(DEFAULT_IMAGE_NAME.withTag("2017-CU12"));
```

Comment:
any later version for test here ?

Reply:
I use 2022 version for test, this tag is used in the original container test package. I keep it. And use the latest version in our test code.

Reply:
Should I change it? I suggest keep it.

Reply:
sorry but what does "this tag is used in the original container test package" mean ?

Reply:
do u mean copied from testContainer ? maybe we should just import this ?

Reply:
the all utils are copied from test container and edit a little, so copied from the package

Reply:
So in order to keep the project code style I copied it. But I think you are right, we can edit the version in the test container directly because this code is in our project

Reply:
> do u mean copied from testContainer ? maybe we should just import this Container class? 

I tried but have some bug, conflict with the current project code